MusicNotationKit/MusicNotationKit/Measure.swift

```diff
@@ -46,12 +46,36 @@ public struct Measure: ImmutableMeasure, Equatable {
         noteCount = immutableMeasure.noteCount
         recomputeNoteCollectionIndexes()
     }
-
+	
+	public func note(at index: Int, inSet setIndex: Int) throws -> Note {
+		let noteCollectionIndex = try noteCollectionIndexFromNoteIndex(index, inSet: setIndex)
+		var noteIndex = 0
+		if noteCollectionIndex.tupletIndex != nil {
+			noteIndex = noteCollectionIndex.tupletIndex!
+		}
+		return try notes[setIndex][noteCollectionIndex.noteIndex].note(at: noteIndex)
+	}
+
+	public mutating func replaceNote(at index: Int, inSet setIndex: Int, with note: Note) throws {
+		let noteCollectionIndex = try noteCollectionIndexFromNoteIndex(index, inSet: setIndex)
+		if noteCollectionIndex.tupletIndex == nil {
+			notes[setIndex][noteCollectionIndex.noteIndex] = note
+			return
+		}
+		
+		guard var tuplet = notes[setIndex][noteCollectionIndex.noteIndex] as? Tuplet,
+			let tupletIndex = noteCollectionIndex.tupletIndex else {
+			throw MeasureError.internalError
+		}
```

Comment:
Don't want to explicitly check for `nil`. 

I would rework this in the following way:

```swift
guard let tupletIndex = noteCollectionIndex.tupletIndex else {
    notes[setIndex][noteCollectionIndex.noteIndex] = note
    return
}
guard var tuplet = notes[setIndex][noteCollectionIndex.noteIndex] as? Tuplet else {
    assertionFailure("note collection should be tuplet, but cast failed")
    throw MeasureError.internalError
}
```

Also, I would do an `assertionFailure` with a message to explain what the problem is like all the other internal errors we have.